src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,81 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
```

Comment:
In Stack we use LIFO (Last In First Out), so for optimization, the better way Is use the tail of your Node lists, but it is just advice to you =) And if you remake your code with using tail, your code will  smaller